core/state/statedb.go

```diff
@@ -802,28 +802,48 @@ func (s *StateDB) clearJournalAndRefund() {
 	s.validRevisions = s.validRevisions[:0] // Snapshots can be created without journal entires
 }
 
+type ModifiedAccount struct {
+	Account
+	Storage
+}
+type StateChanges map[common.Address]ModifiedAccount
```

Comment:
Not sure where else these two types would/should go, but was kind of thinking that it may be nice to have them in a different file?